services/QuillLMS/spec/controllers/skills_reports_controller_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe SkillsReportsController do
+  let(:teacher) { create(:teacher) }
+  let(:classroom) { create(:classroom) }
+  let(:concept) { create(:concept) }
+  let(:sub_concept) { create(:concept, parent: concept) }
+
+  before { allow(controller).to receive(:current_user) { teacher } }
+
+  describe '#performance_breakdown_by_student' do
+    let(:valid_params) do
+      {
+        classroom_id: classroom.id,
+        concept_id: concept.id,
+        all_history: false
+      }
+    end
+
+    context 'when user is authorized for classroom' do
+      before do
+        create(:classrooms_teacher, user: teacher, classroom:)
+      end
+
+      context 'with valid concept_id only' do
+        it 'calls SkillsReport::ClassroomScoresByUser with concept_id' do
+          expect(SkillsReport::ClassroomScoresByUser).to receive(:run)
+            .with(classroom_id: classroom.id, concept_id: concept.id, all_history: false)
+            .and_return([])
+
+          get :performance_breakdown_by_student, params: valid_params
+          expect(response).to have_http_status(:ok)
+          expect(JSON.parse(response.body)['data']).to eq([])
```

Comment:
The `JSON.prase(response.body)` call is used in a couple of places, so I'd be tempted to just do `let(:json_response) { JSON.parse(response.body) }` and then do your tests around that value: `expect(json_response['data']).to eq([])`